### PR TITLE
CLI: Run `storybook dev` as part of `storybook init`

### LINKED ISSUE #22871: [Project - Onboarding]: Automate call to `storybook dev <url-params>` after `storybook init`
## Description

The purpose of this task is to automate the execution of the `storybook dev --initial-path /onboarding` command immediately after the storybook init command. Still, this enhancement should only apply to React-based frameworks. This adjustment will reduce the steps required to start a Storybook with custom parameters for React projects, making the development process more efficient.

## Prerequisite

This enhancement is only applicable to frameworks that use React. Other frameworks should not be affected by this change.

## Acceptance Criteria:

- Upon successful execution of storybook init, `storybook dev --initial-path /onboarding` should automatically be called, but only if the initialized project uses a React-based framework.
- The CLI option `--initial-path /onboarding` should be passed as arguments to the storybook dev command.
- In case storybook init fails for any reason, `storybook dev --initial-path /onboarding` should not be called and a meaningful error message should be logged.
- For non-React frameworks, `init` should also run `dev` automatically but without the onboarding flag.
- All existing functionality of storybook init and storybook dev should remain unaffected for React and non-React projects.
- Detection of React-based framework: The solution should accurately determine whether the framework of the initialized project is based on React.

### PR DESCRIPTION
**Pull request by yannbf**
Closes #22871

## What I did

- add --initial-path option to the CLI
- run storybook dev as part of storybook init

## How to test

1. Build all packages
2. Create a new project with the CLI e.g. `./code/lib/cli/bin/index.js init --initial-path=/settings/shortcuts`
3. Expect Storybook to open in that route

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/cli/src/initiate.ts [hunk 1]
```diff
@@ -319,9 +320,29 @@ async function doInitiate(options: CommandOptions, pkg: PackageJson): Promise<vo
...
+      ...options,
+      port: 6006,
+      open: true,
+      quiet: true,
+      // TODO: change this logic to all frameworks once the idea is validated
+      initialPath: isReactProject ? '/onboarding' : undefined,
+    });
+  } else if (projectType === ProjectType.ANGULAR) {
     logger.log('\nTo run your Storybook, type:\n');
     codeLog([`ng run ${installResult.projectName}:storybook`]);
```
Comment (vanessayuenn) [inline code/lib/cli/src/initiate.ts:344]: Can/should we automate this command for angular projects also?

File: code/lib/cli/src/generate.ts [hunk 2]
```diff
@@ -216,6 +216,10 @@ command('dev')
   )
   .option('--force-build-preview', 'Build the preview iframe even if you are using --preview-url')
   .option('--docs', 'Build a documentation-only site using addon-docs')
+  .option(
+    '--initial-path [path]',
```
Comment (valentinpalkovic) [inline code/lib/cli/src/generate.ts:220]: This has to be added to the angular schema as well, if we aim to support this CLI flag there as well. (https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/builders/start-storybook/schema.json, https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/builders/start-storybook/index.ts)

**Review by valentinpalkovic — APPROVED** [review]

File: code/lib/cli/src/initiate.ts (hunk 1 above)
Comment (yannbf) [reply code/lib/cli/src/initiate.ts:344]: Sure thing!

File: code/lib/cli/src/generate.ts (hunk 2 above)
Comment (yannbf) [reply code/lib/cli/src/generate.ts:220]: Thanks!

**Review by shilman — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

File: code/lib/cli/src/initiate.ts [hunk 3]
```diff
@@ -319,9 +320,53 @@ async function doInitiate(options: CommandOptions, pkg: PackageJson): Promise<vo
...
 
-  if (projectType === ProjectType.ANGULAR) {
-    logger.log('\nTo run your Storybook, type:\n');
-    codeLog([`ng run ${installResult.projectName}:storybook`]);
+  const shouldRunDev =
+    projectType !== ProjectType.REACT_NATIVE &&
+    process.env.CI !== 'true' &&
+    process.env.IN_STORYBOOK_SANDBOX !== 'true';
+  if (shouldRunDev) {
+    logger.log('\nRunning Storybook');
```
Comment (yannbf) [inline code/lib/cli/src/initiate.ts:328]: Running Storybook automatically. To run it manually, use the following command: `xyz`

Comment (yannbf) [reply code/lib/cli/src/initiate.ts:328]: @shilman @jonniebigodes here's how it looks like after the change:

For Vite:
image


For Angular Webpack (the most verbose we have):
image


Thoughts?

Comment (shilman) [reply code/lib/cli/src/initiate.ts:328]: Looks great! I'm sure we can clean all this up if we rework the CLI, but for now I'd say ship it!

**Pull request merged**